test/unit/model/jdk-install-test.js

```diff
@@ -134,104 +133,97 @@ describe('JDK installer', function() {
     });
 
     describe('on windows', function() {
-      it('should select openjdk for installation if no java detected', function(done) {
+      it('should select openjdk for installation if no java detected', function() {
         sandbox.stub(Platform, 'getOS').returns('win32');
         let jdk = new JdkInstall(installerDataSvc, 'url', 'jdk8.msi', 'jdk8', 'sha');
         mockDetectedJvm('');
-        return jdk.detectExistingInstall(function() {
+        return jdk.detectExistingInstall().then(()=>{
           expect(jdk.selectedOption).to.be.equal('install');
           expect(jdk.getLocation()).to.be.equal('');
-          done();
         });
       });
 
 
       // FIXME is not the case for JDK 9, because version has different format
-      it('should select openjdk for installation if newer than supported java version detected', function(done) {
+      it('should select openjdk for installation if newer than supported java version detected', function() {
         sandbox.stub(Platform, 'getOS').returns('win32');
         let jdk = new JdkInstall(installerDataSvc, 'url', 'jdk8.msi', 'jdk8', 'sha');
         mockDetectedJvm('1.9.0_1');
-        return jdk.detectExistingInstall(function() {
+        return jdk.detectExistingInstall().then(()=>{
           expect(jdk.selectedOption).to.be.equal('detected');
-          done();
         });
       });
 
-      it('should select openjdk for installation if older than supported java version detected', function(done) {
+      it('should select openjdk for installation if older than supported java version detected', function() {
         sandbox.stub(Platform, 'getOS').returns('win32');
         let jdk = new JdkInstall(installerDataSvc, 'url', 'jdk8.msi', 'jdk8', 'sha');
         mockDetectedJvm('1.7.0_1');
-        return jdk.detectExistingInstall(function() {
+        return jdk.detectExistingInstall().then(()=>{
           expect(jdk.selectedOption).to.be.equal('install');
-          done();
         });
       });
 
-      it('should select openjdk for installation if location for java is not found', function(done) {
+      it('should reject openjdk if location for java is not found', function() {
         sandbox.stub(Platform, 'getOS').returns('win32');
         let jdk = new JdkInstall(installerDataSvc, 'url', 'jdk8.msi', 'jdk8', 'sha');
         mockDetectedJvm('1.8.0_1', '');
-        return jdk.detectExistingInstall(function() {
-          expect(jdk.selectedOption).to.be.equal('install');
-          done();
-        });
+        try {
```

Comment:
This test was correct, if for some reason java is in path, but java location still cannot be detected, then detection method promise is going to be rejected. That means installer should mark jdk for installation. So the old test should be just convertd to use promise and that would be it.